packages/simcore-sdk/src/simcore_sdk/node_ports_common/r_clone.py

```diff
@@ -2,25 +2,51 @@
 import logging
 import re
 import shlex
-import urllib.parse
+from abc import abstractmethod
+from asyncio.streams import StreamReader
+from collections.abc import AsyncIterator
 from contextlib import asynccontextmanager
 from pathlib import Path
-from typing import AsyncIterator
+from typing import Final
 
 from aiocache import cached
 from aiofiles import tempfile
-from models_library.api_schemas_storage import FileUploadSchema
+from pydantic import AnyUrl
 from pydantic.errors import PydanticErrorMixin
+from servicelib.progress_bar import ProgressBarData
+from servicelib.utils import logged_gather
 from settings_library.r_clone import RCloneSettings
 from settings_library.utils_r_clone import get_r_clone_config
 
-logger = logging.getLogger(__name__)
+S3_RETRIES: Final[int] = 3
+S3_PARALLELISM: Final[int] = 5
 
+_S3_CONFIG_KEY_DESTINATION: Final[str] = "s3-destination"
+_S3_CONFIG_KEY_SOURCE: Final[str] = "s3-source"
 
-class RCloneFailedError(PydanticErrorMixin, RuntimeError):
+_logger = logging.getLogger(__name__)
+
+
+class BaseRCloneLogParser:
+    @abstractmethod
+    async def __call__(self, logs: str) -> None:
+        ...
+
+
+class BaseRCloneError(PydanticErrorMixin, RuntimeError):
+    ...
+
+
+class RCloneFailedError(BaseRCloneError):
     msg_template: str = "Command {command} finished with exception:\n{stdout}"
 
 
+class RCloneFileFoundError(BaseRCloneError):
```

Comment:
that is a bit of a weird error... you might go with RCloneDirectoryNotFoundError instead?